File: rgseaaSLibraryDomain/src/main/java/es/aesan/rgseaa/model/entity/Product.java
package es.aesan.rgseaa.model.entity;


import es.aesan.rgseaa.model.commom.db.AuditedBaseEntity;
import lombok.*;

import java.time.LocalDate;


@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@EqualsAndHashCode(callSuper = false)
public class Product extends AuditedBaseEntity {


    private Long id;

    private Key key;

    private String industryNum;

    private String enrollment;

    private String numProduct;

    private String numRegister;

    private Situation situation;

    private DenominationSale denominationSale;

    private String brand;

    private Entitie competentAuthority;

    private boolean requestDocumentAnnotated;

    private Long industryId;

    private String emailResponsible;

    private Country countryIndustryResponsible;

    private Long provinceIndustryResponsible;

    private Long townIndustryResponsible;

    private String addressIndustryResponsible;

    private Long codePostalResponsible;

    private Country countryManufacturingKey26;

    private Country countryCommercializedKey26;

    private FormPresentation formPresentationKey26;

    private String typePackagingKey26;

    private String observationKey26;

    private String ingredientKey26;

    private String industryNameKey26;

    private String addressKey26;

    private String emailKey26;

    private LocalDate annotatedDateKey26;

    private LocalDate lastDateKey26;

    private Country countryNotificationKey26;

    private String source;

    private String placeLocation;

    private String registrationKey27;

    private String descCountryLocation;

    private Integer countryLocation;

    private String otherCountries;

    private State stateProduct;

    private String ccaaUserCreator;

    private String socialReasonKey26;

    private String productNameIaKey26;

    private String industryPhone;

    private Long substanceType;

    private String phoneIaCode26;

    private LocalDate dateTermination;

    private LocalDate dateDenial;

    private String productType;

    private LocalDate dateRegister27;

    private LocalDate dateUnregister27;

    private LocalDate dateTermination27;

    private String otherCountriesComer;

    private String otherForms;

    private String migration;

    private String counter;

    private String migrationFolder;

    private String nameExcel;

    private LocalDate dateApplication;

    private String formattedMark;

    private Situation idSituationOld;

    private String otherCountryCompanyResp;

    private String otherCountryNotification;

    private String webPage;

    private Country countryKey26;

}

File: rgseaaWeb/src/libs/sdk/product.ts
import {Situation} from "@libs/sdk/situation";
import {Key} from "@libs/sdk/key";
import {Country} from "@libs/sdk/country";
import {DenominationSale} from "@libs/sdk/denomination-sale";
import {Entitie} from "@libs/sdk/entitie";
import {FormPresentation} from "@libs/sdk/form-presentation";
import {Action} from "@libs/sdk/action";
import {AppQuerySource} from "@libs/commons";
import {TypeAction} from "@libs/sdk/type-action";

export interface Product {

    id                              : number;
    key                             : Key | null;
    industryNum                     : string | null;
    enrollment                      : string | null;
    numProduct                      : number | null;
    numRegister                     : number | null;
    situation                       : Situation | null;
    denominationSale                : DenominationSale | null;
    brand                           : string | null;
    competentAuthority              : Entitie | null;
    requestDocumentAnnotated        : boolean | null;

    industryId                      : number | null;
    emailResponsible                : string | null;
    countryIndustryResponsible      : Country | null;
    provinceIndustryResponsible     : string | null;
    townIndustryResponsible         : string | null;
    addressIndustryResponsible      : string | null;
    codePostalResponsible           : string | null;

    countryManufacturingKey26       : Country | null;
    countryCommercializedKey26      : Country | null;
    formPresentationKey26           : FormPresentation | null;
    typePackagingKey26              : string | null;
    observationKey26                : string | null;
    ingredientKey26                 : string | null;
    industryNameKey26               : string | null;
    addressKey26                    : string | null;
    emailKey26                      : string | null;
    annotatedDateKey26              : Date | null;
    lastDateKey26                   : Date | null;
    countryNotificationKey26        : Country | null;
    action                          : Action[] | null;
    numBusiness                     : number | null;
}

export interface ProductCreate {

    // Datos de Producto
    id                  : number | null;
    key                 : Key | null;
    industryNum         : string | null;
    enrollment          : string | null;
    numProduct          : number | null;
    numRegister         : number | null;
    situation           : Situation | null;
    denominationSale    : DenominationSale | null;
    brand               : string | null;
    competentAuthority  :  Entitie | null;
    requestDocumentAnnotated   : boolean | null;

    // Datos Empresa Responsable de la Comercialización
    industryId                          : number | null;
    emailResponsible                    : string | null;
    countryIndustryResponsible          : Country | null;
    provinceIndustryResponsible         : string | null;
    townIndustryResponsible             : string | null;
    addressIndustryResponsible          : string | null;
    codePostalResponsible               : string | null;

    // clave 26
    countryManufacturingKey26           : Country | null;
    countryCommercializedKey26          : Country | null;
    formPresentationKey26               : FormPresentation | null;
    typePackagingKey26                  : string | null;
    observationKey26                    : string | null;
    ingredientKey26                     : string | null;
    industryNameKey26                   : string | null;
    addressKey26                        : string | null;
    emailKey26                          : string | null;
    annotatedDateKey26                  : Date | null;
    lastDateKey26                       : Date | null;
    countryNotificationKey26            : Country | null;
    typeAction                          : TypeAction | null;
    action                              : Action[] | null;

    industryName                        : string | null;

}


export interface ProductFilter extends AppQuerySource {
    industryId?             : string;
    numProduct?             : string;
    numRegister?            : string;
    brand?                  : string;
    ingredientKey26?        : string;
    situation?              : string;
    situationSearch?        : string;
    competentAuthority?     : string;
    competentAuthoritySearch?: string;
    productsEntities?       : string;
}
